### DISCUSSION TITLE
Diff command

### DISCUSSION THREAD

**Discussion by TomWright** [TomWright/dasel#147] [Ideas]
A diff command would be very useful.

From reddit: https://www.reddit.com/r/golang/comments/owbdij/dasel_v1160_added_a_delete_command_meaning_you/h7fq2bp?utm_source=share&utm_medium=web2x&context=3

The diff command should accept 2 documents (in any supported format) and output a standardised diff.

**Comment by vyas-n** [comment]
For anyone looking for this, the following is a workaround using process substitution until this gets added:
```bash
diff <(dasel -f doc1.yaml) <(dasel -f doc2.yaml)
```
```fish
diff (dasel -f doc1.yaml | psub) (dasel -f doc2.yaml | psub)
```

**Comment by TomWright** [comment]
It looks like this could be used to perform the actual diff: https://github.com/sergi/go-diff
Dasel would provide a simple wrapper around that to perform diffs between file formats.

Perhaps it would convert to JSON internally and perform the diff on the text based output to keep things simple.
